File: src/FhirOpenEhrBridge.Domain/Validation/ValidationIssue.cs
namespace FhirOpenEhrBridge.Domain.Validation;

/// <summary>A single problem discovered while validating a payload.</summary>
/// <param name="Severity">How serious the issue is.</param>
/// <param name="Message">Human readable description of the issue.</param>
/// <param name="Location">
/// Optional dotted path to the offending element (e.g. <c>Patient.name[0].family</c>).
/// </param>
public sealed record ValidationIssue(
    ValidationSeverity Severity,
    string Message,
    string? Location = null);

File: java/src/main/java/io/github/hl7ie/fhiropenehrbridge/ValidationIssue.java
package io.github.hl7ie.fhiropenehrbridge;

import com.fasterxml.jackson.annotation.JsonInclude;

/** A single problem found while validating a payload. */
@JsonInclude(JsonInclude.Include.NON_NULL)
public record ValidationIssue(String severity, String message, String location) {

    public static ValidationIssue error(String message, String location) {
        return new ValidationIssue("error", message, location);
    }

    public static ValidationIssue warning(String message, String location) {
        return new ValidationIssue("warning", message, location);
    }
}
